projects/ion/src/lib/tooltip/tooltip.service.ts

```diff
@@ -0,0 +1,131 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { TooltipPosition } from '../core/types';
+
+export type RepositionData = {
+  tooltipLeft: number;
+  tooltipHeight: number;
+  hostPosition: Partial<DOMRect>;
+  screenWidth: number;
+  screenHeight: number;
+};
+
+export type tooltipPositionChecks = {
+  [x in TooltipPosition]?: () => boolean;
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TooltipService {
+  reposition = new Subject();
+  hostPosition: Partial<DOMRect>;
+  screenSize: HTMLElement;
+  tootipCoordinates: DOMRect;
+  elementPadding = 16;
+  elementMaxWidth = 208 + this.elementPadding;
```

Comment:
All attributes that you don't need outside this class should be private.
`reposition` should be public but with readonly, it will prevent that someone else change the value.